### PR TITLE
N average

### LINKED ISSUE #4: Program crashes if trying to plot when a certain channel has no recorded data

### PR DESCRIPTION
**Pull request by ggggggggg**
This adds a spin box for how many traces to average, and makes the y range fields visible by default. 

I realized the y range fields were already implemented (yay!), but I just didn't know how to bring them up.  

Fixes a pain point of using Dastard, which was viewing noise FFTs with chosen ranges and enough averaging.

Fixes #4.

### CODE REVIEW

File: plotwindow.cpp [hunk 1]
```diff
@@ -227,7 +227,7 @@ plotWindow::plotWindow(zmq::context_t *context_in, options *opt, QWidget *parent
 
     // Restore relevant settings.
     mscopeSettings = new QSettings();
-    preferVisibleMinMaxRange = mscopeSettings->value("plots/visibleMinMaxRange",false).toBool();
+    preferVisibleMinMaxRange = mscopeSettings->value("plots/visibleMinMaxRange",true).toBool();
```
Comment (joefowler) [inline plotwindow.cpp]: I think that this setting should be false to save real estate for the plots, and that users should be able to figure out how to expose this Min/Max/Range stuff. But if your example shows that some users aren't as smart as others, then fine: we can do it this way.

File: pulsehistory.cpp [hunk 2]
```diff
@@ -31,243 +31,245 @@ void pulseHistory::clearAllData() {
...
-
-
-///
-/// \brief pulseHistory::size
-/// \return
-///
-int pulseHistory::size() const {
-    return records.size();
-}
+                pulse_peak.reserve(RESERVE);
```
Comment (joefowler) [inline pulsehistory.cpp]: There is some kind of ridiculous indentation problem here. Is it tabs vs spaces? Can you fix it?

Comment (joefowler) [reply pulsehistory.cpp]: See my new commit. I think I fixed it.

File: plotwindow.cpp (hunk 1 above)
Comment (ggggggggg) [reply plotwindow.cpp]: I changed it back to frustrate less intelligent users.

**Pull request merged**